misc/python/materialize/scratch.py

```diff
@@ -226,18 +226,24 @@ def is_ready(i: Instance) -> bool:
     mkrepo(i, git_rev)
 
 
-def mkrepo(i: Instance, rev: str) -> None:
-    mssh(i, "git init --bare materialize/.git")
+def mkrepo(i: Instance, rev: str, init: bool = True, force: bool = False) -> None:
+    if init:
+        mssh(i, "git init --bare materialize/.git")
+
+    cmd: List[str] = [
+        "git",
+        "push",
+        "--no-verify",
+        f"{instance_host(i)}:materialize/.git",
+        # Explicit refspec is required if the host repository is in detached
+        # HEAD mode.
+        "HEAD:refs/heads/scratch",
```

Comment:
It occurs to me: shouldn't this `HEAD` be `git.rev_parse(rev)`? Otherwise we're just hoping that `rev` is an ancestor of the current `HEAD`.

Reply:
I am not quite sure...below when we checkout we do the rev_parse

Reply:
Yeah I'm pretty sure I just got this wrong. Can you hoist the `rev = git.rev_parse(rev)` call up to above this line, and then swap `HEAD` for `rev`?

Reply:
what does `rev_parse` do?